src/Gizmos/axisScaleGizmo.ts

```diff
@@ -146,7 +146,15 @@ export class AxisScaleGizmo extends Gizmo {
 
                 const scalingMatrix = new Matrix();
                 Matrix.ScalingToRef(1 + tmpVector.x, 1 + tmpVector.y, 1 + tmpVector.z, scalingMatrix);
-                this.attachedNode.getWorldMatrix().copyFrom(scalingMatrix.multiply(this.attachedNode.getWorldMatrix()));
+
+                var newScaledMatrix = scalingMatrix.multiply(this.attachedNode.getWorldMatrix());

```

Comment:
Maybe multiplyToRef to a temp matrix instead? This seems to be executed onDrag, which means on each frame when the user drags - This matrix will be GCed after each frame).

Reply:
I'm fixing that !